packages/phpstan-rules/src/DeadCode/UnusedPublicClassConstRule.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\DeadCode;
+
+use Nette\Utils\Arrays;
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\CollectedDataNode;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+use Symplify\PHPStanRules\Collector\ClassConstFetchCollector;
+use Symplify\PHPStanRules\Collector\PublicClassLikeConstCollector;
+use Symplify\RuleDocGenerator\Contract\DocumentedRuleInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Symplify\PHPStanRules\Tests\Rules\DeadCode\UnusedPublicClassConstRule\UnusedPublicClassConstRuleTest
+ */
+final class UnusedPublicClassConstRule implements Rule, DocumentedRuleInterface
```

Comment:
It probably should be `UnusedPublicConstRule`, which should handle constants on classes **and interfaces**. Such naming convention should be used in other places which now refer to "class".

Reply:
I see. In php-parser there are `Const` nodes that cover also non-class-like constants, like:

```php
define('ERROR_LEVEL', 1000);
```

This rule only covers public constants defined inside class/interface.

Reply:
Yes, but rule's name contains unnecessary `Class` infix, so it's misleading... As far as I remember there's concept of `ClassLike` in parser? Maybe it would be better?

BTW. Tests don't cover consts on interfaces.

Reply:
Class like sounds good :+1: Interface const test too :+1:

Reply:
@TomasVotruba I wanted to make a PR with this, but unfortunately even though `nikic/php-parser` has `ClassLike` as a base for `Class_` and `Interface_`, at the same time it has `ClassConstFetch` which is used in your `ClassConstFetchCollector`. I think it would be better to stick to this convention and now I understand why you named it this way 😉 

What do you think? Maybe some phpDoc in the rule and related services informing that it collects both from classes and interfaces? Or break parser's convention and use `ClassLike` in this context?

Reply:
I think it's convention to use class for all such elements, no need to overcomplicate it :)